### PR TITLE
Migrate to sbt-typelevel

### PR DESCRIPTION
**Pull request by armanbilge**

### CODE REVIEW

File: build.sbt
Comment (armanbilge) [reply build.sbt:19]: Actually, we could dial this back to Scala 3.0.2 and merge this today.

**Comment by armanbilge** [comment]
Woops, conflicts...

File: build.sbt
Comment (rossabaker) [reply build.sbt:65]: As an alternative to regular semver?  What's the difference when we're past 0.x?

**Review by rossabaker — CHANGES_REQUESTED** [review]
Seems good to me.  Requesting changes to make sure the Scala 3 gets rolled back to 3.0.2, or else we're deciding for the whole world.

File: build.sbt
Comment (armanbilge) [reply build.sbt:65]: For some reason I thought scalacheck may be using PVP? Happy to be wrong.

Comment (armanbilge) [reply build.sbt:65]: Yeah, the release notes say 1.14.x is not backwards-compatible with anything before it.
https://github.com/typelevel/scalacheck/blob/main/CHANGELOG.markdown#1140-2018-04-22

Comment (rossabaker) [reply build.sbt:65]: Oh, yeah.  The breaking change in a minor version was a driver for a major version bump in libraries as prominent as Cats.  Binary compatibility weighs very heavily on this project, and we shouldn't do that again.

**Comment by rossabaker** [comment]
Does this supersede #860?

**Comment by armanbilge** [comment]
Yes, supersedes https://github.com/typelevel/scalacheck/pull/860 with apologies to @vasilmkd who put a lot of time/effort into it as well. Also closes https://github.com/typelevel/scalacheck/issues/625.

I rolled back to 3.0.2 and removed the Native/Scala3 job. Should be good to go.

**Comment by rossabaker** [comment]
I didn't understand we lose native support on Scala 3 with this rollback.  So our choices are:

1. Keep blocking native support on Scala 3
2. Force an upgrade on all the JVM clients that have stuck to 3.0 (or make them pin scalacheck too)
3. Complicate this further by publshing the JVM on 3.0 and Native on 3.1

?

**Comment by armanbilge** [comment]
Yes, Scala Native requires 3.1. I think we should wait until the new `-Yscala-release` flag arrives in 3.1.2, at which point we can compile all platforms on Scala 3.1 without forcing any downstream to upgrade.

FWIW, https://github.com/scalameta/munit/pull/477 is an important PR blocked by this one, and they are also waiting for the new flag.

**Comment by armanbilge** [comment]
See also https://github.com/portable-scala/sbt-crossproject/issues/130 for why I don't want to mess around with option (3).

**Comment by rossabaker** [comment]
I suppose a fourth option is a prerelease as the next minor version that builds for Scala 3.1, with the assumption that Scala 3.1.2 will precede it.  

As someone almost entirely on Scala 2 and not at all on Scala Native, easy for me to say, but I'm content to wait.

**Comment by armanbilge** [comment]
I'd be happy to get this PR merged as-is, and then we can discuss the Native/3 situation :)

**Review by rossabaker — APPROVED** [review]
I've never been a primary maintainer, so I won't hit the green button, but if nobody objects in the next couple days, I will.

**Comment by armanbilge** [comment]
Thanks!

Gah, conflicts again!

**Comment by rossabaker** [comment]
Sorry, I just merged one that had been approved for months.

**Comment by armanbilge** [comment]
No worries, I'm glad to see activity here :)

**Pull request merged**
